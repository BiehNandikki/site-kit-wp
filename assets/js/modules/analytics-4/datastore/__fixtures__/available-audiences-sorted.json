[
  {
    "audienceSlug": "",
    "audienceType": "USER_AUDIENCE",
    "description": "Description",
    "displayName": "Test Audience",
    "name": "properties/12345/audiences/5"
  },
  {
    "audienceSlug": "new-visitors",
    "audienceType": "SITE_KIT_AUDIENCE",
    "description": "People who visited the site for the first time",
    "displayName": "New visitors",
    "name": "properties/12345/audiences/3"
  },
  {
    "adsPersonalizationEnabled": true,
    "audienceSlug": "returning-visitors",
    "audienceType": "SITE_KIT_AUDIENCE",
    "description": "People who have visited your site at least once before",
    "displayName": "Returning visitors",
    "name": "properties/12345/audiences/4"
  },
  {
    "audienceSlug": "all-users",
    "audienceType": "DEFAULT_AUDIENCE",
    "description": "All users",
    "displayName": "All visitors",
    "name": "properties/12345/audiences/1"
  },
  {
    "audienceSlug": "purchasers",
    "audienceType": "DEFAULT_AUDIENCE",
    "description": "Users who have made a purchase",
    "displayName": "Purchasers",
    "name": "properties/12345/audiences/2"
  }
]
