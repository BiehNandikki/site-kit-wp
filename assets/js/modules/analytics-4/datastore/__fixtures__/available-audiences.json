[
    {
        "name": "properties/437749618/audiences/8004992724",
        "displayName": "All visitors",
        "description": "All users",
        "audienceType": "DEFAULT_AUDIENCE",
        "audienceSlug": "all-users"
    },
    {
        "name": "properties/437749618/audiences/8004992725",
        "displayName": "Purchasers",
        "description": "Users who have made a purchase",
        "audienceType": "DEFAULT_AUDIENCE",
        "audienceSlug": "purchasers"
    },
    {
        "name": "properties/437749618/audiences/8030142001",
        "displayName": "Test Audience",
        "description": "This is the test description for test audience.",
        "audienceType": "SITE_KIT_AUDIENCE",
        "audienceSlug": "returning-visitors"
    }
]
