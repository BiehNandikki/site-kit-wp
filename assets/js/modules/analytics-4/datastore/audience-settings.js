/**
 * `modules/analytics-4` data store: audience settings.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
// import invariant from 'invariant';
// import { isPlainObject } from 'lodash';

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';
import Data from 'googlesitekit-data';
import { MODULES_ANALYTICS_4 } from './constants';
import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
// import { createValidatedAction } from '../../../googlesitekit/data/utils';
import { createReducer } from '../../../googlesitekit/data/create-reducer';
// import { actions as errorStoreActions } from '../../../googlesitekit/data/create-error-store';

// const { receiveError, clearError } = errorStoreActions;

const fetchStoreReducerCallback = createReducer(
	( state, audienceSettings ) => {
		if ( ! state.audienceSettings ) {
			state.audienceSettings = {};
		}

		state.audienceSettings.settings = audienceSettings;
		state.audienceSettings.savedSettings = audienceSettings;
	}
);

const fetchGetAudienceSettingsStore = createFetchStore( {
	baseName: 'getAudienceSettings',
	controlCallback() {
		return API.get(
			'modules',
			'analytics-4',
			'audience-settings',
			{},
			{
				useCache: false,
			}
		);
	},
	reducerCallback: fetchStoreReducerCallback,
} );

const baseInitialState = {
	audienceSettings: undefined,
};

const baseActions = {};

const baseControls = {};

const baseReducer = ( state, { type } ) => {
	switch ( type ) {
		default: {
			return state;
		}
	}
};

const baseResolvers = {
	*getAudienceSettings() {
		const registry = yield Data.commonActions.getRegistry();

		const audienceSettings = registry
			.select( MODULES_ANALYTICS_4 )
			.getAudienceSettings();

		if ( audienceSettings === undefined ) {
			yield fetchGetAudienceSettingsStore.actions.fetchGetAudienceSettings();
		}
	},
};

const baseSelectors = {
	/**
	 * Gets the audience settings.
	 *
	 * @since n.e.x.t
	 *
	 * @param {Object} state Data store's state.
	 * @return {(Object|undefined)} Audience settings; `undefined` if not loaded.
	 */
	getAudienceSettings( state ) {
		return state.audienceSettings;
	},
};

const store = Data.combineStores( fetchGetAudienceSettingsStore, {
	initialState: baseInitialState,
	actions: baseActions,
	controls: baseControls,
	reducer: baseReducer,
	resolvers: baseResolvers,
	selectors: baseSelectors,
} );

export const initialState = store.initialState;
export const actions = store.actions;
export const controls = store.controls;
export const reducer = store.reducer;
export const resolvers = store.resolvers;
export const selectors = store.selectors;

export default store;
