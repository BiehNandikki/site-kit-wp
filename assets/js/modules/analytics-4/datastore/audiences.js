/**
 * `modules/analytics-4` data store: audiences.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';
import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';

const fetchGetAudiencesStore = createFetchStore( {
	baseName: 'getAudiences',
	controlCallback() {
		return API.get(
			'modules',
			'analytics-4',
			'audiences',
			{},
			{
				useCache: false,
			}
		);
	},
	reducerCallback( state, audiences ) {
		return { ...state, audiences };
	},
} );
fetchGetAudiencesStore();
