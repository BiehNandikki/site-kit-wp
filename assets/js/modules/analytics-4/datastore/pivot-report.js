/**
 * `modules/analytics-4` data store: pivotReport.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import invariant from 'invariant';
import { isPlainObject } from 'lodash';

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';
import Data from 'googlesitekit-data';
import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
import { MODULES_ANALYTICS_4 } from './constants';
import { stringifyObject } from '../../../util';
import { isValidDateRange } from '../../../util/report-validation';
import {
	normalizeReportOptions,
	isValidDimensionFilters,
	isValidDimensions,
	isValidMetrics,
	isValidMetricFilters,
	isValidPivots,
} from '../utils';

const fetchGetReportStore = createFetchStore( {
	baseName: 'getPivotReport',
	controlCallback: ( { options } ) => {
		return API.get(
			'modules',
			'analytics-4',
			'pivot-report',
			normalizeReportOptions( options )
		);
	},
	reducerCallback: ( state, report, { options } ) => {
		return {
			...state,
			pivotReports: {
				...state.pivotReports,
				[ stringifyObject( options ) ]: report,
			},
		};
	},
	argsToParams: ( options ) => {
		return { options };
	},
	validateParams: ( { options } = {} ) => {
		invariant(
			isPlainObject( options ),
			'options for Analytics 4 report must be an object.'
		);
		invariant(
			isValidDateRange( options ),
			'Either date range or start/end dates must be provided for Analytics 4 report.'
		);

		const {
			metrics,
			dimensions,
			dimensionFilters,
			metricFilters,
			pivots,
			orderby,
			limit,
		} = normalizeReportOptions( options );

		invariant(
			metrics.length,
			'Requests must specify at least one metric for an Analytics 4 report.'
		);
		invariant(
			isValidMetrics( metrics ),
			'metrics for an Analytics 4 report must be either a string, an array of strings, an object, an array of objects, or a mix of strings and objects. Objects must have a "name" property. Metric names must match the expression ^[a-zA-Z0-9_]+$.'
		);
		invariant(
			isValidPivots( pivots ),
			'pivots for an Analytics 4 report must be an array of objects. Each object must have a "fieldNames" property and a "limit".'
		);

		if ( orderby ) {
			invariant(
				Array.isArray( orderby ),
				'orderby for an Analytics 4 pivot report must be passed within a pivot.'
			);
		}
		if ( limit ) {
			invariant(
				typeof limit === 'number',
				'limit for an Analytics 4 pivot report must be passed within a pivot.'
			);
		}

		if ( dimensions ) {
			invariant(
				isValidDimensions( dimensions ),
				'dimensions for an Analytics 4 report must be either a string, an array of strings, an object, an array of objects, or a mix of strings and objects. Objects must have a "name" property.'
			);
		}

		if ( dimensionFilters ) {
			invariant(
				isValidDimensionFilters( dimensionFilters ),
				'dimensionFilters for an Analytics 4 report must be a map of dimension names as keys and dimension values as values.'
			);
		}

		if ( metricFilters ) {
			invariant(
				isValidMetricFilters( metricFilters ),
				'metricFilters for an Analytics 4 report must be a map of metric names as keys and filter value(s) as numeric fields, depending on the filterType.'
			);
		}
	},
} );

const baseInitialState = {
	pivotReports: {},
};

const baseResolvers = {
	*getPivotReport( options = {} ) {
		const registry = yield Data.commonActions.getRegistry();
		const existingReport = registry
			.select( MODULES_ANALYTICS_4 )
			.getPivotReport( options );

		// If there is already a report loaded in state, consider it fulfilled
		// and don't make an API request.
		if ( existingReport ) {
			return;
		}

		yield fetchGetReportStore.actions.fetchGetPivotReport( options );
	},
};

const baseSelectors = {
	/**
	 * Gets an Analytics pivot report for the given options.
	 *
	 * @since n.e.x.t
	 *
	 * @param {Object}         state                      Data store's state.
	 * @param {Object}         options                    Options for generating the report.
	 * @param {string}         options.startDate          Required. Start date to query report data for as YYYY-mm-dd.
	 * @param {string}         options.endDate            Required. End date to query report data for as YYYY-mm-dd.
	 * @param {Array.<string>} options.metrics            Required. List of metrics to query.
	 * @param {Array.<string>} options.dimensions         Required. List of dimensions to group results by.
	 * @param {Array.<string>} options.pivots             Required. List of pivots to structure the report data into columns.
	 * @param {Object}         [options.dimensionFilters] Optional. Map of dimension filters for filtering options on a dimension. Default an empty object.
	 * @param {Object}         [options.metricFilters]    Optional. Map of metric filters for filtering options on a metric. Default an empty object.
	 * @param {string}         [options.url]              Optional. URL to get a report for only this URL. Default an empty string.
	 * @return {(Array.<Object>|undefined)} An Analytics report; `undefined` if not loaded.
	 */
	getPivotReport( state, options ) {
		const { pivotReports } = state;

		return pivotReports[ stringifyObject( options ) ];
	},
};

const store = Data.combineStores( fetchGetReportStore, {
	initialState: baseInitialState,
	resolvers: baseResolvers,
	selectors: baseSelectors,
} );

export const initialState = store.initialState;
export const actions = store.actions;
export const controls = store.controls;
export const reducer = store.reducer;
export const resolvers = store.resolvers;
export const selectors = store.selectors;

export default store;
