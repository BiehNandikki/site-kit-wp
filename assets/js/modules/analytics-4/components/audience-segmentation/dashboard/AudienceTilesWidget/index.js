/**
 * AudienceTilesWidget component.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';
import { useEffect, useState } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { useDispatch, useSelect } from 'googlesitekit-data';
import whenActive from '../../../../../../util/when-active';
import { MODULES_ANALYTICS_4 } from '../../../../datastore/constants';
import AudienceTiles from './AudienceTiles';
import { useInView } from '../../../../../../hooks/useInView';

function AudienceTilesWidget( { Widget, WidgetNull } ) {
	const availableAudiences = useSelect( ( select ) => {
		const audiences = select( MODULES_ANALYTICS_4 ).getAvailableAudiences();
		return audiences?.map( ( audience ) => audience.name );
	} );
	const configuredAudiences = useSelect( ( select ) =>
		select( MODULES_ANALYTICS_4 ).getConfiguredAudiences()
	);

	const [ availableAudiencesSynced, setAvailableAudiencesSynced ] =
		useState( false );
	const { maybeSyncAvailableAudiences } = useDispatch( MODULES_ANALYTICS_4 );

	const inView = useInView();
	useEffect( () => {
		if ( inView && ! availableAudiencesSynced ) {
			maybeSyncAvailableAudiences();
			setAvailableAudiencesSynced( true );
		}
	}, [ inView, availableAudiencesSynced, maybeSyncAvailableAudiences ] );

	const hasMatchingAudience = configuredAudiences?.some( ( audience ) =>
		availableAudiences?.includes( audience )
	);

	if ( ! hasMatchingAudience ) {
		return <WidgetNull />;
	}

	return (
		<AudienceTiles
			Widget={ Widget }
			widgetLoading={
				! availableAudiencesSynced ||
				! availableAudiences ||
				! configuredAudiences
			}
		/>
	);
}

AudienceTilesWidget.propTypes = {
	Widget: PropTypes.elementType.isRequired,
	WidgetNull: PropTypes.elementType.isRequired,
};

export default whenActive( { moduleName: 'analytics-4' } )(
	AudienceTilesWidget
);
