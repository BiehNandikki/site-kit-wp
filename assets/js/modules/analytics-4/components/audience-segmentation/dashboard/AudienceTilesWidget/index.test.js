/**
 * AudienceTilesWidget component tests.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import AudienceTilesWidget from '.';
import { render } from '../../../../../../../../tests/js/test-utils';
import {
	createTestRegistry,
	muteFetch,
	provideModuleRegistrations,
	provideModules,
	unsubscribeFromAll,
} from '../../../../../../../../tests/js/utils';
import { getWidgetComponentProps } from '../../../../../../googlesitekit/widgets/util';
import { availableAudiences } from '../../../../datastore/__fixtures__';
import { MODULES_ANALYTICS_4 } from '../../../../datastore/constants';

describe( 'AudienceTilesWidget', () => {
	let registry;

	const { Widget, WidgetNull } = getWidgetComponentProps(
		'analyticsAudienceTiles'
	);

	const auduenceSettingsRegExp = new RegExp(
		'^/google-site-kit/v1/modules/analytics-4/data/audience-settings'
	);

	beforeEach( () => {
		registry = createTestRegistry();
		provideModules( registry, [
			{
				active: true,
				connected: true,
				slug: 'analytics-4',
			},
		] );
		provideModuleRegistrations( registry );
		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {} );
	} );

	afterEach( () => {
		unsubscribeFromAll( registry );
		jest.clearAllMocks();
	} );

	it( 'should not render when availableAudiences and configuredAudiences are not loaded', () => {
		muteFetch( auduenceSettingsRegExp );

		const { container } = render(
			<AudienceTilesWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
			{
				registry,
			}
		);

		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should not render when availableAudiences is not loaded', () => {
		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetAudienceSettings( {
			configuredAudiences: [ 'properties/12345/audiences/1' ],
			isAudienceSegmentationWidgetHidden: false,
		} );

		const { container } = render(
			<AudienceTilesWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
			{
				registry,
			}
		);

		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should not render when configuredAudiences is not loaded', () => {
		muteFetch( auduenceSettingsRegExp );

		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setAvailableAudiences( availableAudiences );

		const { container } = render(
			<AudienceTilesWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
			{
				registry,
			}
		);

		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should not render when there is no available audience', () => {
		registry.dispatch( MODULES_ANALYTICS_4 ).setAvailableAudiences( [] );

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetAudienceSettings( {
			configuredAudiences: [ 'properties/12345/audiences/9' ],
			isAudienceSegmentationWidgetHidden: false,
		} );

		const { container } = render(
			<AudienceTilesWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
			{
				registry,
			}
		);

		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should not render when there is no configured audience', () => {
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setAvailableAudiences( availableAudiences );

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetAudienceSettings( {
			configuredAudiences: [],
			isAudienceSegmentationWidgetHidden: false,
		} );

		const { container } = render(
			<AudienceTilesWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
			{
				registry,
			}
		);

		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should not render when configuredAudiences is null (not set)', () => {
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setAvailableAudiences( availableAudiences );

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetAudienceSettings( {
			configuredAudiences: null,
			isAudienceSegmentationWidgetHidden: false,
		} );

		const { container } = render(
			<AudienceTilesWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
			{
				registry,
			}
		);

		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should not render when there is no matching audience', () => {
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setAvailableAudiences( availableAudiences );

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetAudienceSettings( {
			configuredAudiences: [ 'properties/12345/audiences/9' ],
			isAudienceSegmentationWidgetHidden: false,
		} );

		const { container } = render(
			<AudienceTilesWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
			{
				registry,
			}
		);

		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should render render when configured audience is matching available audiences', async () => {
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setAvailableAudiences( availableAudiences );

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetAudienceSettings( {
			configuredAudiences: [ 'properties/12345/audiences/1' ],
			isAudienceSegmentationWidgetHidden: false,
		} );

		const { container, waitForRegistry } = render(
			<AudienceTilesWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
			{
				registry,
			}
		);

		await waitForRegistry();

		expect( container ).not.toBeEmptyDOMElement();
	} );

	it( 'should render when all configured audiences are matching available audiences', async () => {
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setAvailableAudiences( availableAudiences );

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetAudienceSettings( {
			configuredAudiences: [
				'properties/12345/audiences/1',
				'properties/12345/audiences/3',
			],
			isAudienceSegmentationWidgetHidden: false,
		} );

		const { container, waitForRegistry } = render(
			<AudienceTilesWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
			{
				registry,
			}
		);

		await waitForRegistry();

		expect( container ).not.toBeEmptyDOMElement();
	} );

	it( 'should render when some configured audiences are matching available audiences', async () => {
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setAvailableAudiences( availableAudiences );

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetAudienceSettings( {
			configuredAudiences: [
				'properties/12345/audiences/1',
				'properties/12345/audiences/9',
			],
			isAudienceSegmentationWidgetHidden: false,
		} );

		const { container, waitForRegistry } = render(
			<AudienceTilesWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
			{
				registry,
			}
		);

		await waitForRegistry();

		expect( container ).not.toBeEmptyDOMElement();
	} );
} );
