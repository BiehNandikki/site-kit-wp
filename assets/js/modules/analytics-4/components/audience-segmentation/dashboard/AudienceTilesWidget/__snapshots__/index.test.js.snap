// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudienceTilesWidget should not render audiences that are not available (archived) 1`] = `
<div>
  <div
    class="googlesitekit-widget googlesitekit-widget--analyticsAudienceTiles googlesitekit-widget--no-padding googlesitekit-widget-audience-tiles"
  >
    <div
      class="googlesitekit-widget__body"
    >
      <div
        class="mdc-tab-bar googlesitekit-widget-audience-tiles__tabs"
        dir="ltr"
        role="tablist"
      >
        <div
          class="mdc-tab-scroller"
        >
          <div
            class="mdc-tab-scroller__scroll-area mdc-tab-scroller__scroll-area--scroll"
            style="margin-bottom: 0px;"
          >
            <div
              class="mdc-tab-scroller__scroll-content"
            >
              <button
                aria-label="All visitors"
                aria-selected="true"
                class="mdc-tab mdc-tab--active"
                role="tab"
                tabindex="0"
              >
                <span
                  class="mdc-tab__content"
                >
                  All visitors
                  <span
                    class="googlesitekit-info-tooltip"
                  >
                    <svg />
                  </span>
                </span>
                <span
                  class="mdc-tab-indicator mdc-tab-indicator--active"
                >
                  <span
                    class="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
                    style=""
                  />
                </span>
                <div
                  class="mdc-tab__ripple mdc-ripple-upgraded--background-focused"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="googlesitekit-widget-audience-tiles__body"
      >
        <div
          class="googlesitekit-widget googlesitekit-widget--analyticsAudienceTiles googlesitekit-widget--no-padding"
        >
          <div
            class="googlesitekit-widget__body"
          >
            <div
              class="googlesitekit-audience-segmentation-tile"
            >
              <div
                class="googlesitekit-audience-segmentation-tile__zero-data-container"
              >
                <div
                  class="googlesitekit-audience-segmentation-tile__zero-data-content"
                >
                  <svg />
                  <p
                    class="googlesitekit-audience-segmentation-tile__zero-data-description"
                  >
                    Site Kit is collecting data for this group.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AudienceTilesWidget should render when all configured audiences are matching available audiences 1`] = `
<div>
  <div
    class="googlesitekit-widget googlesitekit-widget--analyticsAudienceTiles googlesitekit-widget--no-padding googlesitekit-widget-audience-tiles"
  >
    <div
      class="googlesitekit-widget__body"
    >
      <div
        class="mdc-tab-bar googlesitekit-widget-audience-tiles__tabs"
        dir="ltr"
        role="tablist"
      >
        <div
          class="mdc-tab-scroller"
        >
          <div
            class="mdc-tab-scroller__scroll-area mdc-tab-scroller__scroll-area--scroll"
            style="margin-bottom: 0px;"
          >
            <div
              class="mdc-tab-scroller__scroll-content"
            >
              <button
                aria-label="All visitors"
                aria-selected="true"
                class="mdc-tab mdc-tab--active"
                role="tab"
                tabindex="0"
              >
                <span
                  class="mdc-tab__content"
                >
                  All visitors
                  <span
                    class="googlesitekit-info-tooltip"
                  >
                    <svg />
                  </span>
                </span>
                <span
                  class="mdc-tab-indicator mdc-tab-indicator--active"
                >
                  <span
                    class="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
                    style=""
                  />
                </span>
                <div
                  class="mdc-tab__ripple mdc-ripple-upgraded--background-focused"
                />
              </button>
              <button
                aria-label="New visitors"
                aria-selected="false"
                class="mdc-tab"
                role="tab"
                tabindex="-1"
              >
                <span
                  class="mdc-tab__content"
                >
                  New visitors
                  <span
                    class="googlesitekit-info-tooltip"
                  >
                    <svg />
                  </span>
                </span>
                <span
                  class="mdc-tab-indicator"
                >
                  <span
                    class="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
                  />
                </span>
                <div
                  class="mdc-tab__ripple"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="googlesitekit-widget-audience-tiles__body"
      >
        <div
          class="googlesitekit-widget googlesitekit-widget--analyticsAudienceTiles googlesitekit-widget--no-padding"
        >
          <div
            class="googlesitekit-widget__body"
          >
            <div
              class="googlesitekit-audience-segmentation-tile"
            >
              <div
                class="googlesitekit-audience-segmentation-tile__zero-data-container"
              >
                <div
                  class="googlesitekit-audience-segmentation-tile__zero-data-content"
                >
                  <svg />
                  <p
                    class="googlesitekit-audience-segmentation-tile__zero-data-description"
                  >
                    Site Kit is collecting data for this group.
                  </p>
                  <p
                    class="googlesitekit-audience-segmentation-tile__zero-data-description"
                  >
                    You can hide this group until data is available.
                  </p>
                  <button
                    class="googlesitekit-cta-link googlesitekit-audience-segmentation-tile-hide-cta googlesitekit-cta-link--secondary googlesitekit-cta-link--link-button"
                  >
                    <span
                      class="googlesitekit-icon-wrapper"
                      style="margin-right: 5px;"
                    >
                      <svg />
                    </span>
                    <span
                      class="googlesitekit-cta-link__contents"
                    >
                      Temporarily hide
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AudienceTilesWidget should render when configured audience is matching available audiences 1`] = `
<div>
  <div
    class="googlesitekit-widget googlesitekit-widget--analyticsAudienceTiles googlesitekit-widget--no-padding googlesitekit-widget-audience-tiles"
  >
    <div
      class="googlesitekit-widget__body"
    >
      <div
        class="mdc-tab-bar googlesitekit-widget-audience-tiles__tabs"
        dir="ltr"
        role="tablist"
      >
        <div
          class="mdc-tab-scroller"
        >
          <div
            class="mdc-tab-scroller__scroll-area mdc-tab-scroller__scroll-area--scroll"
            style="margin-bottom: 0px;"
          >
            <div
              class="mdc-tab-scroller__scroll-content"
            >
              <button
                aria-label="All visitors"
                aria-selected="true"
                class="mdc-tab mdc-tab--active"
                role="tab"
                tabindex="0"
              >
                <span
                  class="mdc-tab__content"
                >
                  All visitors
                  <span
                    class="googlesitekit-info-tooltip"
                  >
                    <svg />
                  </span>
                </span>
                <span
                  class="mdc-tab-indicator mdc-tab-indicator--active"
                >
                  <span
                    class="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
                    style=""
                  />
                </span>
                <div
                  class="mdc-tab__ripple mdc-ripple-upgraded--background-focused"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="googlesitekit-widget-audience-tiles__body"
      >
        <div
          class="googlesitekit-widget googlesitekit-widget--analyticsAudienceTiles googlesitekit-widget--no-padding"
        >
          <div
            class="googlesitekit-widget__body"
          >
            <div
              class="googlesitekit-audience-segmentation-tile"
            >
              <div
                class="googlesitekit-audience-segmentation-tile__zero-data-container"
              >
                <div
                  class="googlesitekit-audience-segmentation-tile__zero-data-content"
                >
                  <svg />
                  <p
                    class="googlesitekit-audience-segmentation-tile__zero-data-description"
                  >
                    Site Kit is collecting data for this group.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AudienceTilesWidget should render when some configured audiences are matching available audiences 1`] = `
<div>
  <div
    class="googlesitekit-widget googlesitekit-widget--analyticsAudienceTiles googlesitekit-widget--no-padding googlesitekit-widget-audience-tiles"
  >
    <div
      class="googlesitekit-widget__body"
    >
      <div
        class="mdc-tab-bar googlesitekit-widget-audience-tiles__tabs"
        dir="ltr"
        role="tablist"
      >
        <div
          class="mdc-tab-scroller"
        >
          <div
            class="mdc-tab-scroller__scroll-area mdc-tab-scroller__scroll-area--scroll"
            style="margin-bottom: 0px;"
          >
            <div
              class="mdc-tab-scroller__scroll-content"
            >
              <button
                aria-label="All visitors"
                aria-selected="true"
                class="mdc-tab mdc-tab--active"
                role="tab"
                tabindex="0"
              >
                <span
                  class="mdc-tab__content"
                >
                  All visitors
                  <span
                    class="googlesitekit-info-tooltip"
                  >
                    <svg />
                  </span>
                </span>
                <span
                  class="mdc-tab-indicator mdc-tab-indicator--active"
                >
                  <span
                    class="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
                    style=""
                  />
                </span>
                <div
                  class="mdc-tab__ripple mdc-ripple-upgraded--background-focused"
                />
              </button>
              <button
                aria-label="New visitors"
                aria-selected="false"
                class="mdc-tab"
                role="tab"
                tabindex="-1"
              >
                <span
                  class="mdc-tab__content"
                >
                  New visitors
                  <span
                    class="googlesitekit-info-tooltip"
                  >
                    <svg />
                  </span>
                </span>
                <span
                  class="mdc-tab-indicator"
                >
                  <span
                    class="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
                  />
                </span>
                <div
                  class="mdc-tab__ripple"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="googlesitekit-widget-audience-tiles__body"
      >
        <div
          class="googlesitekit-widget googlesitekit-widget--analyticsAudienceTiles googlesitekit-widget--no-padding"
        >
          <div
            class="googlesitekit-widget__body"
          >
            <div
              class="googlesitekit-audience-segmentation-tile"
            >
              <div
                class="googlesitekit-audience-segmentation-tile__zero-data-container"
              >
                <div
                  class="googlesitekit-audience-segmentation-tile__zero-data-content"
                >
                  <svg />
                  <p
                    class="googlesitekit-audience-segmentation-tile__zero-data-description"
                  >
                    Site Kit is collecting data for this group.
                  </p>
                  <p
                    class="googlesitekit-audience-segmentation-tile__zero-data-description"
                  >
                    You can hide this group until data is available.
                  </p>
                  <button
                    class="googlesitekit-cta-link googlesitekit-audience-segmentation-tile-hide-cta googlesitekit-cta-link--secondary googlesitekit-cta-link--link-button"
                  >
                    <span
                      class="googlesitekit-icon-wrapper"
                      style="margin-right: 5px;"
                    >
                      <svg />
                    </span>
                    <span
                      class="googlesitekit-cta-link__contents"
                    >
                      Temporarily hide
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
